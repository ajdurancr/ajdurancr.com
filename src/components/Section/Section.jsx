import React from 'react'
import propTypes from 'prop-types'
import get from 'lodash.get'

import { Component } from '..'

const Section = ({ components }) => (
  <>
    {components && components.map((component) => (
      <Component key={get(component, 'sys.id')} {...component} />
    ))}
  </>
)

Section.defaultProps = {
  components: [],
}

Section.propTypes = {
  components: propTypes.arrayOf(propTypes.object),
}

export default Section
