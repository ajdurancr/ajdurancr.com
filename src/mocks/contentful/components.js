const childComponent = {
  sys: {
    id: '4gfjlQfm3x9ShZKO8gdYqx',
    contentType: {
      sys: {
        id: 'component',
      },
    },
  },
  fields: {
    name: 'ChildComponent',
    contentType: 'Test Component',
    componentDescription: 'This is a child component',
  },
}

export const component = {
  sys: {
    id: '6neuJJsr2mbdKli0VjWnCn',
    contentType: {
      sys: {
        id: 'component',
      },
    },
  },
  fields: {
    name: 'TestComponent',
    contentType: 'Test Component',
    componentDescription: 'This is a test component',
    props: [
      {
        sys: {
          id: '74gEMnRKSOxejniOwjkHYN',
          contentType: {
            sys: {
              id: 'propText',
            },
          },
        },
        fields: {
          name: 'title',
          value: 'This is a test title',
        },
      },
      {
        sys: {
          id: '3hwiQ7OcvmctEdQPye6exu',
          contentType: {
            sys: {
              id: 'propComponents',
            },
          },
        },
        fields: {
          name: 'children',
          value: [
            childComponent,
          ],
        },
      },
      {
        sys: {
          id: '1RezpPBXf2buGYIccqQw0i',
          contentType: {
            sys: {
              id: 'propMarkdown',
            },
          },
        },
        fields: {
          name: 'markdownField',
          value: 'Test value',
        },
      },
    ],
  },
}

export default [component]
