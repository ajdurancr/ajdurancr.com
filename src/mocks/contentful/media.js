export default {
  title: 'Media - GitHub Logo',
  description: 'Github Logo',
  file: {
    url: '//images.ctfassets.net/52uxn1pv2fou/5Yt9H0zn6ewSMnuvzh7V6P/3d6a253c1dce0250c70ef3a52c169872/github.jpg',
    details: {
      size: 11536,
      image: {
        width: 400,
        height: 400,
      },
    },
    fileName: 'GitHub_filename',
    contentType: 'image/jpeg',
  },
}
