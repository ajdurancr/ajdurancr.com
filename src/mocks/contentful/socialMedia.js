import socialMediaLinks from './socialMediaLinks'

export default {
  sys: {
    id: '3Aj6YhWjxQsZvrJbmPbm0i',
    contentType: {
      sys: { id: 'component' },
    },
  },
  fields: {
    name: 'Social Media',
    contentType: 'Social Media',
    props: [
      {
        sys: {
          id: '4mQeFUKhFefDEGBEoJG4C8',
          contentType: {
            sys: { id: 'propComponents' },
          },
        },
        fields: {
          name: 'socialMediaLinks',
          value: socialMediaLinks,
        },
      },
    ],
  },
}
