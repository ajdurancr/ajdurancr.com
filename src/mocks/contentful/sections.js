import components from './components'

export const section = {
  sys: {
    id: 'sectionId',
  },
  fields: {
    components,
    name: 'Main Section',
  },
}

export default [section]
