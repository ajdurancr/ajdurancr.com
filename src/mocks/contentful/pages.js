import sections from './sections'

const pages = [
  {
    fields: {
      name: 'Home',
      url: '/',
      sections,
    },
  },
  {
    fields: {
      name: 'About',
      url: '/about',
      sections,
    },
  },
  {
    fields: {
      name: 'Blog',
      url: '/blog',
      sections,
    },
  },
  {
    fields: {
      name: 'Portfolio',
      url: '/portfolio',
      sections,
    },
  },
  {
    fields: {
      name: 'Resume',
      url: '/resume',
      sections: undefined,
    },
  },
  {
    fields: {
      name: 'Education',
      url: '/resume/education',
      sections,
    },
  },
  {
    fields: {
      name: 'Work Experience',
      url: '/resume/work-experience',
      sections,
    },
  },
]

export const pagesMap = pages.reduce((pagesAcc, page) => ({
  ...pagesAcc,
  [page.fields.url]: page,
}), {})

export default pages
