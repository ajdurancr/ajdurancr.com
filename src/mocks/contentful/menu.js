import { pagesMap } from './pages'

export const childSubmenu = [
  {
    fields: {
      alias: undefined,
      name: 'Menu - DCR.IO Resume / Education',
      page: pagesMap['/resume/education'],
      submenu: undefined,
      url: undefined,
    },
  },
  {
    fields: {
      alias: undefined,
      name: 'Menu - DCR.IO Resume / Work Experience',
      page: pagesMap['/resume/work-experience'],
      submenu: undefined,
      url: undefined,
    },
  },
]

export const submenu = [
  {
    fields: {
      alias: undefined,
      name: 'Menu - DCR.IO Home Item',
      page: pagesMap['/'],
      submenu: undefined,
      url: undefined,
    },
  },
  {
    fields: {
      alias: undefined,
      name: 'Menu - DCR.IO Blog',
      page: pagesMap['/blog'],
      submenu: undefined,
      url: undefined,
    },
  },
  {
    fields: {
      alias: undefined,
      name: 'Menu - DCR.IO Resume',
      submenu: childSubmenu,
      page: pagesMap['/resume'],
      url: undefined,
    },
  },
  {
    fields: {
      alias: undefined,
      name: 'Menu - DCR.IO Portfolio',
      page: pagesMap['/portfolio'],
      submenu: undefined,
      url: undefined,
    },
  },
  {
    fields: {
      alias: undefined,
      name: 'Menu - DCR.IO About',
      page: pagesMap['/about'],
      submenu: undefined,
      url: undefined,
    },
  },
  {
    fields: {
      alias: undefined,
      name: 'Menu - DCR.IO External link',
      submenu: undefined,
      page: undefined,
      url: 'http://www.google.com',
    },
  },
  {
    fields: {
      alias: 'Google',
      name: 'Menu - DCR.IO External link with alias',
      submenu: undefined,
      page: undefined,
      url: 'http://www.google.com',
    },
  },
]

export default {
  fields: {
    name: 'Top Menu',
    submenu,
  },
}
