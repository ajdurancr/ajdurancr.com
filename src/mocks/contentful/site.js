import pages from './pages'
import menu from './menu'

export default {
  fields: {
    name: 'DURANCR.IO',
    id: 1,
    menu,
    pages,
  },
}
