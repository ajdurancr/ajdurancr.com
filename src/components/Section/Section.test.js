import React from 'react'
import { shallow } from 'enzyme'

import Section from './Section'
import { section } from '../../mocks/contentful/sections'

describe('Section', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Section {...section} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with default props', () => {
    const wrapper = shallow(<Section name="Test Section" />)

    expect(wrapper).toMatchSnapshot()
  })
})
