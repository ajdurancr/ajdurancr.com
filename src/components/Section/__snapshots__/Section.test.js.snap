// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section renders correctly 1`] = `<Fragment />`;

exports[`Section renders correctly with default props 1`] = `<Fragment />`;
